src/conn/conn_dhandle.c

```diff
@@ -64,16 +70,28 @@ __conn_dhandle_config_set(WT_SESSION_IMPL *session)
     switch (dhandle->type) {
     case WT_DHANDLE_TYPE_BTREE:
         WT_ERR(__wt_strdup(session, WT_CONFIG_BASE(session, file_meta), &dhandle->cfg[0]));
+        WT_ASSERT(session, dhandle->meta_base == NULL);
+        /* Overwrite any checkpoint info. */
+        cfg[1] = "checkpoint=()";
```

Comment:
I'm not sure why this is necessary -- collapse doesn't discard anything, it just gets you a single set of items? IOW, doesn't the subsequent discard take care of any existing checkpoint info?

Reply:
No. This specific piece of code was the most difficult to work out in fact. The `wt_config_merge` call below with `checkpoint=,...` in the `strip` arg will not actually strip out real checkpoint information. That code requires and only strips out specific elements when a subcategory is involved. So for an actual checkpoint record, that would need the exact checkpoint name and each piece. For example, `checkpoint=(WiredTigerCheckpoint.37=(addr=))`. That wasn't going to happen. I tried for a while to get the merge/strip code to remove the entire category, but it started getting messy in lower levels of the config parsing and I never got it working. Given how I can simply make it work with the two calls, those code changes did not seem worth debugging or taking more time.